postgresql-11.5/src/bin/psql/commandbteq.c

```diff
@@ -117,7 +128,7 @@ HandleDotCmds(BteqScanState scan_state,
 
     /* Parse off the command name */
     cmd = bteq_scan_dot_command(scan_state);
-
+    cat_symbols(cmd, "; ");
     if (PQstatus(pset.db) == CONNECTION_BAD && strcasecmp(cmd, "logon") != 0 &&
```

Comment:
I believe that to resolve the issue described above you will need to update this part to accept set width before logon.

Another note related to this: Nearly all BTEQ commands are going before LOGON as well, so you just need to disable executing SQL queries.

Reply:
I believe these will be the last changes, and we will be able to merge PR into the master finally! :)